core/src/debuginfo/source_file.rs

```diff
@@ -368,18 +378,24 @@ impl miette::SourceCode for SourceFileRef {
 /// * The byte offsets of every line in the file, for use in looking up line/column information
 #[derive(Clone)]
 pub struct SourceContent {
+    /// The language identifier for this source file
+    language: Box<str>,
     /// The path (or name) of this file
-    path: Arc<str>,
+    uri: Uri,
     /// The underlying content of this file
-    content: Box<str>,
+    content: String,
     /// The byte offsets for each line in this file
-    line_starts: Box<[ByteIndex]>,
+    line_starts: Vec<ByteIndex>,
+    /// The document version
+    version: i32,
 }
 
 impl fmt::Debug for SourceContent {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         f.debug_struct("SourceContent")
-            .field("path", &self.path)
```

Comment:
nit: I like to start these functions with

```rs
let Self { ...<fields>... } = self;
// ...
```
so that we get a compile-time error next time that struct changes. Could apply to a bunch of other methods (e.g. the implementation of `hash()`)